### PR TITLE
Fix terraform plan file should not be deleted

### PR DESCRIPTION
**Pull request by samtholiya**
issue: https://linear.app/cloudposse/issue/DEV-2825/atmos-terraform-show-deletes-planfile

## what

* Stop deleting planfile after terraform show

## why
We should not delete the terraform plan file if the user uses show to check his results.

# Test

Steps:
1. `cd examples/quick-start-simple`
2. `../../build/atmos terraform plan -s dev station`
3.  `../../build/atmos terraform show -s dev station --json dev-station.planfile --skip-init`

Verified if the file `./examples/quick-start-simple/components/terraform/weather/dev-station.planfile` still exists.

Before this change: deleted

After this change: present

## references

issue: https://linear.app/cloudposse/issue/DEV-2825/atmos-terraform-show-deletes-planfile

conversation: https://sweetops.slack.com/archives/C031919U8A0/p1733924783112799

### CODE REVIEW

**Comment by osterman** [comment]
I think you've identified the culprit here causing the problem however I think that our approach is wrong. Rather than by default deleting files for a virtually any command, which is an unexpected behavior, especially for command like running help, I think we should instead identify when if ever we should delete files automatically. We have a separate command called `atmos terraform clean` that handles this. 

In other words, I think what I'm proposing is we just remove this deletion altogether.

**Comment by osterman** [comment]
@aknysh what are your thoughts about this?

**Comment by aknysh** [comment]
> @aknysh what are your thoughts about this?

we had disabled the deletion altogether once, but then people started complaining that Atmos leaves a lot of files behind :)
we re-enabled it again

I agree that the best approach would be to only clean after the commands that actually generate the files

All help commands get executed before this part of the code, they don't delete any files
